src/checkout-resource.js

```diff
@@ -67,6 +68,30 @@ class CheckoutResource extends Resource {
       .then(handleCheckoutMutation('checkoutCreate', this.graphQLClient));
   }
 
+  /**
+   * Updates a checkout.
```

Comment:
does this update and persist new values sent, or merge with existing values?

Reply:
It updates with the new values sent. So if previously `customAttributes: [{key: "Key', value: "Value"}]` existed, and the new input was `customAttributes: [{key: "DifferentKey", value: "DifferentValue"}]`, the new value would be persisted, and the old discarded.

Reply:
Shouldn't this be `Update's a checkout's custom attributes` or something more specific?

Reply:
If call one was `customAttributes: [{key: "keyOne", value: "valueOne"}]` and call two was `customAttributes: [{key: "keyTwo", value: "valueTwo"}]`

Would the resultant `customAttributes` have both `keyOne` and `keyTwo` or just `keyTwo`? If it's the latter, `replaces checkout custom attributes` would be the better descriptor. Either way, `updates a checkout` is way to vague when we're specifically talking about a checkout's custom attributes.

Reply:
You can also update `allowPartialAddress` and `note`. Maybe it can be [checkoutAttributesUpdate](https://help.shopify.com/api/custom-storefronts/storefront-api/reference/input_object/checkoutattributesupdateinput )

Reply:
The description on the graphql side is `Updates the attributes of a checkout.`

Reply:
which is also vague. Line items are attributes of a checkout, but they're not updatable here. Also, what was the answer to my question 3 comments up @StefanSlehta ?

Reply:
The resultant value of `customAttributes` would be just `keyTwo`, however, you can still update only a `note`, without providing `customAttributes` and it wouldn't delete the ones that are currently on the `checkout`. Maybe we could say `Updates or replaces checkout's attributes ` ?

Reply:
we should probably be more specific than just `a checkout's attributes`. this specifically updates `a checkouts `customAttributes and/or notes` field. right? attributes that exist on checkout are significantly more than what can be updated wit this mutation.

Reply:
updates or replaces is ambiguous. this explicitly replaces either one or both of those fields.